src/operators.jl

```diff
@@ -442,3 +442,43 @@ This is equivalent to [`induced_subgraph`](@ref)`(g, neighborhood(g, v, d, dir=d
 with respect to `v` (i.e. `:in` or `:out`).
 """
 egonet(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = g[neighborhood(g, v, d, dir=dir)]
+
+
+"""
+    merge_vertices!(g::Graph, vs::Array{Int})
+
+Combines vertices specified in `vs` into single vertex whose
+index will be the lowest value in `vs`.
+
+All edges running to vertices in `vs` will connect to new
+merged vertex.
+
+Returns an Array{Int} with new vertex values for
+"""
+
+function merge_vertices!(g::Graph, vs::Array{Int})
+
+    allunique(vs) || error("Vertices in vs must be unique")
+
+    vertex_conversions = collect(1:nv(g))
+
+    # Use lowest value as new vertex id.
+    v0 = minimum(vs)
+    filter!(x->x != v0, vs)
+
+    for v in vs
+        # Copy over all connections without adding self-loops
+        for n in neighbors(g, v)
+            if v0 != n
+                add_edge!(g, v0, n)
+            end
+        end
+
+        # Now remove vertex
+        rem_vertex!(g, v)
+        vertex_conversions[v] = v0
+        vertex_conversions[(v+1):end] = vertex_conversions[(v+1):end] - 1
```

Comment:
I think this might cause `O(n*length(vs))` performance because you are updating the `vertex_conversions` every time around the loop `length(vs)` times. Is there a better way to store this information? I think you can compute the final value of `vertex_conversions[i]` based on the number of elements in `vs` that is less that `i`.